Source/WebCore/platform/network/curl/OpenSSLHelper.cpp

```diff
@@ -303,17 +303,79 @@ static void getSubjectAltName(const X509* x509, Vector<String>& dnsNames, Vector
             else if (value->d.iPAddress->length == 16) {
                 StringBuilder ipAddress;
                 for (int i = 0; i < 8; i++) {
-                    ipAddress.append(makeString(hex(data[0] << 8 | data[1], 4)));
+                    ipAddress.append(makeString(hex(data[0] << 8 | data[1], 4, Lowercase)));
                     if (i != 7)
                         ipAddress.append(":");
                     data += 2;
                 }
-                ipAddresses.append(ipAddress.toString());
+                String ipv6 = getIPv6(ipAddress.toString());
```

Comment:
Rather than appending integers to a string just to be parsed again, why don't you pass a std::array<uint16_t, 8> as a parameter?

Reply:
Addressed in https://github.com/WebKit/WebKit/pull/3791/commits/f8bae992ac33b2a826fce129d8884429633ede96